### PR TITLE
[refactored][l]: refactored entire index.js to use different lib and …

### PR DESCRIPTION
**Pull request by Mikanebu**
- refactored entire `index.js` to use library `puppeteer` to test graphs and tables and `cheerio` to traverse HTML
- added appending statuses into `status.csv` 
- updated headers in `status.csv`
  - removed `data_loading_time`, `zip_content`, `first_byte_load_time `, `html_load_time`
  - added new headers `script_loading_time`, `layout_loading_time`
- updated `status.csv`
- added test sample `gdp-uk` and `gdp-us`

### CODE REVIEW

**Review by zelima — APPROVED** [review]
@Mikanebu This looks good, but small comment re PR description for future: 
* please provide reasonable title written by you and not first X number of letters of your first commit message, so that reviewer can easily understand what PR is about. Eg: "Use cheerio instead of  ..."
* Provide small description what is PR about and why did you make that changes - "Refactored to use cheerio" - OK that's nice, but why? what was wrong with the previous approach? 
* Is there issue for that? If yes reference it in PR description, if no it's even more important to provide well understandable reason why you are making changes
* commit messages, they can be more descriptive:
  * " ...  to use different lib ..." - what lib? different than what?
  * `[fixed]` - provides no information about what is commit about - it should describe what commit is about or what the changes are made for in one word Eg: `[dependencies]` would  be much clear message than `[fixed]`

File: index.js [hunk 1]
```diff
@@ -1,132 +1,146 @@
...
 const fetch = require('node-fetch')
 
-class CheckWebsite {
-  constructor(rows) {
-    // TODO: File.rows should do this for us ...
-    this.headers = rows.shift()
-    this.statuses = rows.map(row => {
-      return lodash.zipObject(this.headers, row)
-    })
+var newLine= "\r\n"
```
Comment (zelima) [inline index.js]: let

File: index.js [hunk 2]
```diff
@@ -1,132 +1,146 @@
...
   const res = await fetch(url)
   return res
 }
 
-const datapackageJson = async () => {
-  const url = 'https://pkgstore.datahub.io/core/finance-vix/latest/datapackage.json'
-  const res = await fetch(url)
-  return res
+const writeToCSV = async (fields, toCsv) => {
+  var csv = json2csv(toCsv) + newLine;
```
Comment (zelima) [inline index.js]: let

File: index.js [hunk 3]
```diff
@@ -1,132 +1,146 @@
...
+    if ( datapackageLink.text() === 'Datapackage.json') {
+      let datapackageUrl = baseUrl + datapackageLink[0].attribs.href
+      const dpUrl = await resourceLinks(datapackageUrl)
+      statuses.datapackage_json = dpUrl.status + ':' + dpUrl.statusText
+    }
+  }
+  
+  // append row into csv file
+  let fields = ["id","name","page_status","page_title","dataset_title","readme","csv_links","csv_preview_links","json_links","zip_links","datapackage_json","total_load_time","first_byte_load_time","html_load_time","data_load_time","zip_content","tables","graphs"]
+  var toCsv = {
```
Comment (zelima) [inline index.js]: let

File: index.js [hunk 4]
```diff
@@ -1,132 +1,146 @@
...
+    let datasetTitle = $('.showcase-page-header').find('h1').text()
+    if (datasetTitle.includes(dp.title)) {
+      statuses.dataset_title = 'OK'
+    } else {
+      statuses.dataset_title = 'NOT OK'
+    }
+    
+    // readme
+    let readme = $('.inner_container').find('#readme').text()
+    if (readme === 'Read me') {
```
Comment (zelima) [inline index.js:106]: Is "Read me" fixed? what if it is READEM or readm etc...

**Review by zelima — CHANGES_REQUESTED** [review]
@Mikanebu Forgot this:  Please provide instructions on how to run this in README

File: index.js (hunk 4 above)
Comment (Mikanebu) [reply index.js:106]: @zelima this is fixed

File: README.md [hunk 5]
```diff
@@ -1 +1,28 @@
-# frontend-functional-tests
\ No newline at end of file
+Functional testing for frontend
+
+## Usage
+
+To run the tests, use the following command
+```
+node index.js
```
Comment (zelima) [inline README.md]: Can we make this `npm run ...` as well

File: README.md [hunk 6]
```diff
@@ -1 +1,28 @@
...
+```
+
+All results will be saved in `status.csv` with the following headers:
+`id,name,page_status,page_title,dataset_title,readme,csv_links,csv_preview_links,json_links,zip_links,datapackage_json,total_load_time,script_laod_time,layout_load_time,tables,graphs`
```
Comment (zelima) [inline README.md]: Think this would look better if they are bullet points

File: README.md [hunk 7]
```diff
@@ -1 +1,28 @@
...
+
+If you want to add datasets to test, add dataset into dictionary in `index.js`
```
Comment (zelima) [inline README.md]: Can't we load this from csv or some other document? Would be good to implement in future. We could test all core packages in that way

File: index.js [hunk 8]
```diff
@@ -1,30 +1,28 @@
...
+const puppeteer = require('puppeteer');
 
-var newLine= "\r\n"
+
+let newLine= "\r\n"
 const datasetsToTest = [
   {
     "owner": "core",
-    "name": "finance-vix"
+    "name": "gdp-uk"
```
Comment (zelima) [inline index.js]: what was wrong with finance-vix?

File: index.js [hunk 9]
```diff
@@ -123,24 +141,43 @@ datasetsToTest.forEach(async dataset => {
...
-  }
+    // page loading time 
+    const loadTime = await pageLoadTime(showcaseUrl)
+    statuses.total_load_time = loadTime.TaskDuration
+    statuses.script_laod_time = loadTime.ScriptDuration
+    statuses.layout_load_time = loadTime.LayoutDuration
+    
+    // graphs
+    const graphC = $('.svg-container').css()
+    if (graphC.position === 'relative' && graphC.width === '670px' && graphC.height === '450px') {
```
Comment (zelima) [inline index.js:164]: are you sure this check is enough? is this always the case, Eg what if we change dataset?

File: README.md (hunk 5 above)
Comment (Mikanebu) [reply README.md]: ok

File: README.md (hunk 6 above)
Comment (Mikanebu) [reply README.md]: @zelima you mean, all the headers?

File: README.md (hunk 7 above)
Comment (Mikanebu) [reply README.md]: We can create another csv file, and get from there, it is worth it?

File: index.js (hunk 9 above)
Comment (Mikanebu) [reply index.js:164]: @zelima @anuveyatsu suggested to test this part, it is fixed

File: README.md (hunk 6 above)
Comment (Mikanebu) [reply README.md]: fixed

File: README.md (hunk 7 above)
Comment (Mikanebu) [reply README.md]: @zelima we can do it later, according to the issue we only need to test couple datasets: https://github.com/datahq/pm/issues/42

Comment (zelima) [reply README.md]: @Mikanebu Let's open issue for that and do it later

**Review by zelima — APPROVED** [review]

**Comment by zelima** [comment]
@Mikanebu one more thing: please delete branches that you worked on after merge (unless you really need them later)

**Pull request merged**